### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by bbenne10**
I've gone ahead and reworked some of our README.
In particular, I want to address #138 here, but I felt I should take the time to clarify some other parts too.

They're all nicely squashed up into singular commits, so we can just cherry pick only d3651bd if the rest of the PR is overreach.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -258,41 +275,6 @@ nix_direnv_watch_file mod1.nix mod2.nix
 use flake
 ```
 
-## Shell integration
-
```
Comment (Mic92) [inline README.md:262]: ## Shell integration

See the [wiki](https://github.com/nix-community/nix-direnv/wiki/Shell-integration) for helpers to quickly setup a direnv setup in a new project.

File: README.md [hunk 2]
```diff
@@ -258,41 +275,6 @@ nix_direnv_watch_file mod1.nix mod2.nix
 use flake
 ```
```
Comment (Mic92) [inline README.md:277]:
```suggestion

## Shell integration

See the [wiki](https://github.com/nix-community/nix-direnv/wiki/Shell-integration) for helpers to quickly setup a direnv setup in a new project.
```

**Comment by bbenne10** [comment]
I wonder if using the wiki is a good idea for other bits in the readme?
In particular, the snippet on storing direnv layout directory outside the project directory.

I think the idea is both good and very helpful, but it is not nix-direnv specific.
Maybe it is better placed in the direnv wiki?

**Comment by Mic92** [comment]
`direnv_layout_dir` might be documented in direnv.

**Pull request merged**